### PR TITLE
[xcode14] Bump to Xcode 14

### PR DESCRIPTION
**Pull request by dalexsoto**
We should probably consider a merge commit for this PR so we can revert the revert in the future for BackgroundAssets

* Apple removed BackgroundAssets 
* Apple removed macOS 13.0 SDK
* Apple Removed MacCat 16.0 SDK

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

**Review by chamons — APPROVED** [review]

File: tests/xtro-sharpie/api-annotations-dotnet/macOS-BackgroundAssets.todo [hunk 1]
```diff
@@ -0,0 +1,30 @@
+!missing-enum! BADownloadState not bound
```
Comment (mandel-macaque) [inline tests/xtro-sharpie/api-annotations-dotnet/macOS-BackgroundAssets.todo:1]: So we are adding a macOS todo yet removing the binding that were there, why is that? Can't we just simply add it to mac os x and remove the todo or clean the todo accordingly?

File: tools/common/Frameworks.cs [hunk 2]
```diff
@@ -276,7 +276,6 @@ public Framework Find (string framework)
 
 					{ "ScreenCaptureKit", "ScreenCaptureKit", 12,3 },
 
-					{ "BackgroundAssets", "BackgroundAssets", 13,0 },
```
Comment (mandel-macaque) [inline tools/common/Frameworks.cs:279]: Leave it for macOS, not for the other platforms. Looks to me we are doing a step fwd and 2 steps back with this.

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are unrelated (#15939).

**Pull request merged**